tuf/api/metadata.py

```diff
@@ -3,9 +3,31 @@
 
 """TUF role metadata model.
 
-This module provides container classes for TUF role metadata, including methods
-to read and write from and to file, perform TUF-compliant metadata updates, and
-create and verify signatures.
+This module contains low-level API through container classes for TUF role
+metadata. The API aims to provide:
+- Safe de/serialization of metadata to and from files
+- Access to and modification of signed metadata content
+- Signing metadata and verifying signatures
+
+Each of the top level metadata roles is an instance of the Metadata[T] class
+where the "signed" portion of each of the roles (or the "T") is an instance
+of one of the classes:
+- Root (See https://theupdateframework.readthedocs.io/en/latest/api/tuf.api.metadata.html#tuf.api.metadata.Root)
```

Comment:
I am not sure how do we want to add those links: links that could be copied from a source code editor and pasted in the browser or formatted links according to the readthedocs documentation: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-sections-in-the-same-document

Reply:
I played a lot about how I can beautify the content following the readthedocs format.
@jku what do you think?

Reply:
Yeah... no really good options here. What you did is otherwise great (the fact that it works in source is nice), but we also plan to publish docs for released versions and you can't just link to latest from there.

I think the viable options are to not link at all or to use the anchor references (that won't work in source code). Neither is a great solution.

Reply:
I think if we move the top-level metadata classes before the helper classes and do not add links could be fine.